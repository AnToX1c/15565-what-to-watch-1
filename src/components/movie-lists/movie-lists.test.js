import React from "react";
import renderer from "react-test-renderer";
import MovieLists from "./movie-lists";

const mock = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    imgsrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    imgsrc: `img/bohemian-rhapsody.jpg`
  },
  {
    id: 3,
    title: `Macbeth`,
    imgsrc: `img/macbeth.jpg`
  }
];

it(`MovieLists page correctly renders`, () => {
  const tree = renderer
    .create(<MovieLists films={mock}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
