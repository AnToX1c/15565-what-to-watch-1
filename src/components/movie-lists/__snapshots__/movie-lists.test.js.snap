// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieLists page correctly renders 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      height="175"
      loop={true}
      muted={true}
      poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      height="175"
      loop={true}
      muted={true}
      poster="img/bohemian-rhapsody.jpg"
      width="280"
    >
      <source
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <video
      height="175"
      loop={true}
      muted={true}
      poster="img/macbeth.jpg"
      width="280"
    >
      <source
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
</div>
`;
